<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DebugDialog</class>
 <widget class="QMainWindow" name="DebugDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1512</width>
    <height>1144</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>RTED RuntimeSystem Debugger</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="RoseCodeEdit" name="codeEdit1"/>
      <widget class="RoseCodeEdit" name="codeEdit2"/>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1512</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionEditorSettings"/>
   </widget>
   <widget class="QMenu" name="menuDebug">
    <property name="title">
     <string>Debug</string>
    </property>
    <addaction name="actionResume"/>
    <addaction name="actionSingleStep"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuDebug"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="dockWidget_2">
   <property name="windowIcon">
    <iconset>
     <normaloff>:/icons/document-properties.png</normaloff>:/icons/document-properties.png</iconset>
   </property>
   <property name="windowTitle">
    <string>Stack Variables</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="0" column="0">
      <widget class="QLineEdit" name="txtStackFilter">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item row="0" column="1" rowspan="2">
      <widget class="PropertyTreeWidget" name="propStack"/>
     </item>
     <item row="1" column="0">
      <widget class="QListView" name="lstStack"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget_3">
   <property name="windowIcon">
    <iconset>
     <normaloff>:/icons/document-properties.png</normaloff>:/icons/document-properties.png</iconset>
   </property>
   <property name="windowTitle">
    <string>Heap Variables</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QLineEdit" name="txtHeapFilter"/>
     </item>
     <item row="0" column="1" rowspan="2">
      <widget class="PropertyTreeWidget" name="propHeap"/>
     </item>
     <item row="1" column="0">
      <widget class="QListView" name="lstHeap"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget_4">
   <property name="windowIcon">
    <iconset>
     <normaloff>:/icons/document-properties.png</normaloff>:/icons/document-properties.png</iconset>
   </property>
   <property name="windowTitle">
    <string>Memory Locations</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4">
    <layout class="QGridLayout" name="gridLayout_3">
     <item row="0" column="0">
      <widget class="QLineEdit" name="txtMemFilter"/>
     </item>
     <item row="0" column="1" rowspan="2">
      <widget class="PropertyTreeWidget" name="propMem"/>
     </item>
     <item row="1" column="0">
      <widget class="QListView" name="lstMem"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="editorToolbar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <property name="windowTitle">
    <string>Messages</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QListWidget" name="lstMessages"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="debugToolbar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextBesideIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionSingleStep"/>
   <addaction name="actionResume"/>
  </widget>
  <action name="actionQuit">
   <property name="icon">
    <iconset>
     <normaloff>:/icons/application-exit.png</normaloff>:/icons/application-exit.png</iconset>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset>
     <normaloff>:/icons/document-save.png</normaloff>:/icons/document-save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset>
     <normaloff>:/icons/document-open.png</normaloff>:/icons/document-open.png</iconset>
   </property>
   <property name="text">
    <string>Open...</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="icon">
    <iconset>
     <normaloff>:/icons/document-save-as.png</normaloff>:/icons/document-save-as.png</iconset>
   </property>
   <property name="text">
    <string>Save as...</string>
   </property>
  </action>
  <action name="actionEditorSettings">
   <property name="icon">
    <iconset>
     <normaloff>:/icons/document-properties.png</normaloff>:/icons/document-properties.png</iconset>
   </property>
   <property name="text">
    <string>Editor Settings...</string>
   </property>
  </action>
  <action name="actionSingleStep">
   <property name="icon">
    <iconset>
     <normaloff>:/redo.png</normaloff>:/redo.png</iconset>
   </property>
   <property name="text">
    <string>Single Step</string>
   </property>
  </action>
  <action name="actionResume">
   <property name="icon">
    <iconset>
     <normaloff>:/debug-resume.png</normaloff>:/debug-resume.png</iconset>
   </property>
   <property name="text">
    <string>Resume</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PropertyTreeWidget</class>
   <extends>QTreeView</extends>
   <header>PropertyTreeWidget.h</header>
  </customwidget>
  <customwidget>
   <class>RoseCodeEdit</class>
   <extends>QFrame</extends>
   <header>RoseCodeEdit.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>DebugDialog</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>403</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
